apps/arena/lib/arena/game_updater.ex

```diff
@@ -674,6 +679,36 @@ defmodule Arena.GameUpdater do
     %{game_state | players: players}
   end
 
+  defp activate_traps(game_state) do
+    now = System.monotonic_time(:millisecond)
+
+    activated_traps =
+      Enum.filter(game_state.traps, fn {_trap_id, trap} ->
+        trap.activate_at < now && trap.aditional_info.status == :PREPARED
+      end)
+
+    {game_state, traps} =
+      Enum.reduce(activated_traps, {game_state, game_state.traps}, fn {trap_id, trap}, {game_state_acc, traps_acc} ->
+        game_state = Trap.do_mechanics(game_state_acc, trap, trap.aditional_info.mechanics)
+        trap = put_in(trap, [:aditional_info, :status], :TRIGGERED)
+        {game_state, Map.put(traps_acc, trap_id, trap)}
+      end)
+
+    put_in(game_state, [:traps], traps)
+  end
+
+  defp remove_expired_traps(game_state) do
+    now = System.monotonic_time(:millisecond)
+
+    remaining_traps =
+      Enum.filter(game_state.traps, fn {_trap_id, trap} ->
+        trap.remove_at > now
```

Comment:
Rather than depending on time what do you think about checking the trap state to remove traps? you could remove every triggered trap. If you decid to do this you should check the execution order of these functions in the main tick method

Reply:
This sounds good